pkg/gitrepo/repo.go

```diff
@@ -303,6 +308,77 @@ func (r *Repo) ResolveVersion(ctx context.Context, ref string) (string, error) {
 	return pseudoVersion, nil
 }
 
+// GetProjectVersion retrieves version stored in project's Go source code.
+// It looks up the value of variable `version` in `pkg/project/project.go` file
+// on master branch.
+func (r *Repo) GetProjectVersion() (string, error) {
```

Comment:
I have a feeling this doesn't belong here. This is very codebase specific and it's not related to the git repository itself. My idea is:

- Provide generic method like `(r *Repo) GetFile(ctx context.Context, path string) ([]byte, error)`. This method should also have `IsNotFound` handler. (I can see this method being useful in many other places)
- Have the rest of the code in architect where we'd use it for `architect helm template`.

Reply:
Make sense, I was also mitigated on having this code here.

I'll split this: keeping a `GetFile` method here and moving the parsing logic to architect.

Reply:
* Change this to `GetFileContent`
* Added handling for file not found
* Added tests
* Moved parsing logic to architect https://github.com/giantswarm/architect/pull/369

PTALA